ecs_task.tf

```diff
@@ -1,38 +1,168 @@
+    },
+    { "name" = "DD_VERSION"
+      "value" = "0.0.1"
+    },
+    { "name" = "DD_APM_DD_URL"
+      "value" = "https://trace.agent.datadoghq.com" 
+    },
+    { "name" = "DD_APM_ENABLED"
+      "value" = "true" 
+    },
+    { "name" = "DD_APM_NON_LOCAL_TRAFFIC"
+      "value" = "true" 
+    },
+    { "name" = "DD_DOCKER_ENV_AS_TAGS"
+      "value" = "true" 
+    },
+    { "name" = "DD_DOCKER_LABELS_AS_TAGS"
+      "value" = "true" 
+    },
+    { "name" = "ECS_FARGATE"
+      "value" = "true" 
+    },
+    { "name" = "DD_SITE"
+      "value" = "datadoghq.com" 
+    },
+    { "name" = "DD_USE_PROXY_FOR_CLOUD_METADATA"
+      "value" = "true" 
+    }
+  ]
+
+    # --- Envoy container environment variables --- #
+    envoy_env_vars = [
+    { "name" = "APPMESH_RESOURCE_ARN"
+      "value" = "arn:aws:appmesh:us-east-1:${data.aws_caller_identity.current.id}:mesh/${var.app_mesh_name}@${var.app_mesh_account_id}/virtualNode/vn-${var.app_name}-${var.env_name}-{BG_COLOR}" 
+    },
+    { "name" = "ENABLE_ENVOY_DATADOG_TRACING"
+      "value" = "true" 
+    },
+    { "name" = "ENVOY_LOG_LEVEL"
+      "value" = "off" 
+    },
+    { "name" = "DATADOG_TRACER_PORT"
+      "value" = "8126" 
+    }
+  ]
+
+    # --- App container definitino --- #
+    app_task = {
+            name = var.app_name,
+            image = var.app_container_image,
+            cpu = var.task_definition_cpu,
+            memory = var.task_definition_memory,
+            essential = true,
+            #environment = concat(local.app_env_vars,var.app_environment_variables)
+            #environment = local.app_env_vars
+            #environment = [{"name": "environment", "value": "${local.app_env_vars}"}]
+            #environment = jsonencode(local.app_env_vars)
+            "environment": [
+            {"name": "VARNAME", "value": "VARVAL"}
+             ],
+            secrets = var.app_container_secrets,
+            taskRoleArn = aws_iam_role.ecs_task_execution_role.arn,
+            portMappings = [
                 {
-                    protocol = "tcp"
                     containerPort = var.app_container_port
-                },
-                {
-                    protocol = "tcp"
                     hostPort = var.app_container_port
                 }
-            ]
+            ],
             logConfiguration = {
                 logDriver = "awslogs"
                 options = {
                     awslogs-group = var.aws_cloudwatch_log_group_name
                     awslogs-region = data.aws_region.current.id
                     awslogs-stream-prefix = "${var.app_name}-logs"
                 }
-            }
+            },
             dependsOn = [{
                 containerName = "envoy"
                 condition = "HEALTHY"
             }]
             
         }
     
+    # --- Envoy container definitino --- #
     envoy_task = {        
             name = "envoy"
             image = "public.ecr.aws/appmesh/aws-appmesh-envoy:v1.24.0.0-prod"
```

Comment:
Maybe we should set this as variable with default value so if a new version is out we can easly update the image without releasing a new module